app/assets/javascripts/application.js

```diff
@@ -81,7 +82,7 @@ function onContentLoad() {
 
   tfm.i18n.intl.ready.then(function() {
     var tz = jstz.determine();
-    $.cookie('timezone', tz.name(), {
+    Cookies.set('timezone', tz.name(), {
```

Comment:
In other places we use `tfm` to pull in code from webpack. If we do that, I think we can avoid the Ruby dependency on `js_cookie_rails`.

I actually wonder why we need the timezone cookie. Some of my digging.

It was added in ad998ce7262fdf6cd052d62e3e089f9525ba5c89. There it has `set_timezone` in `app/controllers/concerns/application_shared.rb` but that was moved to `app/controllers/concerns/foreman/controller/timezone.rb` in d1e18585baf05ab9530b6548b2f8873f067393b6. That makes me think it's still used. A quick look on the internet tells me this design is still a best practice.

Having said that: ` tfm.i18n.intl.ready` looks to be coming from webpack code so you can also consider moving the whole cookie setting code to webpack.

Reply:
its used in rubys set_timezone.
I moved it to webpack